src/lib.rs

```diff
@@ -60,79 +61,27 @@ struct DispatchMap {
     next_seq: u64,
 }
 
+impl DispatchMap {
+    fn new() -> Self {
+        Self {
+            map: HashMap::new(),
+            next_seq: 0,
+        }
+    }
+}
+
 impl Client {
     /// Connect to hub.
     ///
     /// Waits for handshake, and optionally for authentication if an auth key is provided.
     pub async fn connect(rpc_addr: SocketAddr, auth_key: Option<&str>) -> RPCResult<Self> {
         let (tx, rx) = std::sync::mpsc::channel();
+        let dispatch = Arc::new(Mutex::new(DispatchMap::new()));
+        let dispatch_connection = Arc::clone(&dispatch);
 
-        let dispatch = Arc::new(Mutex::new(DispatchMap {
-            map: HashMap::new(),
-            next_seq: 0,
-        }));
-
+        ClientConnection::spawn(rpc_addr, rx, dispatch_connection)?;
```

Comment:
I think in this case it could be shortened to:
```suggestion
        ClientConnection::spawn(rpc_addr, rx, dispatch.clone())?;
```

Reply:
Fixed.